### PR TITLE
MGMT-3970 Make the iso editor scratch space base dir configurable

### PR DESCRIPTION
**Pull request by carbonin**
Before this change we would unconditionally extract isos to a
random directory in /tmp and would create new iso files directly
in /tmp

Now that base directory is configurable with the WORKSPACE_BASE_DIR
environment variable. The idea is that we may want this to be a PV
so we are not using the container filesystem.

### CODE REVIEW

**Comment by carbonin** [comment]
Leaving this as WIP for now until I can fully deploy and test it.

**Comment by romfreiman** [comment]
@djzager

**Review by djzager — APPROVED** [review]
Looks like a reasonable change to me.

/lgtm

**Comment by djzager** [comment]
Counting on the WIP being the only necessary hold on this PR :sunglasses:

**Comment by carbonin** [comment]
It seems that this won't fully solve our problem.
This PR will cause us to extract the iso and write the new iso to the specified location, but [go-diskfs uses scratch space as well](https://github.com/diskfs/go-diskfs/blob/d9890cb7fbfd2f6eeae0df021de89342dc189669/filesystem/iso9660/iso9660.go#L81-L83).
And the location for that isn't currently configurable. It will always be created in `/tmp`

This PR is still an improvement, but I will need to make another change to go-diskfs to make the scratch space fully configurable.
The question is, does this make sense to do anyway for now?

Also, this is already configurable, but if we want everything out of `/tmp` we'll also need to set the [`ISO_CACHE_DIR` env var](https://github.com/openshift/assisted-service/blob/3bc73c1259d48b711077f5ef7de0c1fde8d066d9/internal/bminventory/inventory.go#L103).

**Comment by carbonin** [comment]
Opened https://github.com/diskfs/go-diskfs/issues/100 to see if we can change the temp dir handling in go-diskfs

**Comment by carbonin** [comment]
This is no longer a work in progress .. I tested it out end to end. But I do have some open questions, so I'll put a hold on and remove WIP.

If someone is unconcerned my the issues I raised in the last two comments feel free to remove the hold :smile: 

/hold

**Comment by carbonin** [comment]
Opened https://github.com/diskfs/go-diskfs/pull/101 to deal with the issues with the iso workspace

**Comment by carbonin** [comment]
Eh, I guess I'll add the changes to take advantage of https://github.com/diskfs/go-diskfs/pull/101 now that it's merged.

/hold

**Comment by carbonin** [comment]
So, now if you want to change the directory used for iso-related things you'll need to set two env vars:
```
      ISO_WORKSPACE_BASE_DIR:        /data
      ISO_CACHE_DIR:                 /data/cache
```

**Review by djzager — APPROVED** [review]
/lgtm

File: internal/isoutil/isoutil.go [hunk 1]
```diff
@@ -141,43 +139,17 @@ func (h *installerHandler) Create(outPath string, volumeLabel string) error {
 	}
 
 	d.LogicalBlocksize = 2048
-	fspec := disk.FilesystemSpec{Partition: 0, FSType: filesystem.TypeISO9660, VolumeLabel: volumeLabel}
+	fspec := disk.FilesystemSpec{
+		Partition:   0,
+		FSType:      filesystem.TypeISO9660,
+		VolumeLabel: volumeLabel,
+		WorkDir:     h.workDir,
```
Comment (danielerez) [inline internal/isoutil/isoutil.go:146]: don't we still need to clean the work dir?

Comment (carbonin) [reply internal/isoutil/isoutil.go:146]: Not if we call finalize.

I actually added test cases to ensure the directory was removed.

**Pull request merged**